src/worker/core/season/doAwards.hockey.ts

```diff
@@ -0,0 +1,316 @@
+
+const getPlayerInfoOffense = (p: PlayerFiltered): AwardPlayer => {
+	return {
+		pid: p.pid,
+		name: p.name,
+		tid: p.tid,
+		abbrev: p.abbrev,
+		keyStats: p.currentStats.keyStats,
+		pts: p.currentStats.pts,
+	};
+};
+
+const getPlayerInfoDefense = (p: PlayerFiltered): AwardPlayerDefense => {
+	return {
+		pid: p.pid,
+		name: p.name,
+		tid: p.tid,
+		abbrev: p.abbrev,
+		tk: p.currentStats.tk,
+		gaa: p.currentStats.gaa,
+	};
+};
+
+const getTopPlayersOffense = (
+	options: GetTopPlayersOptions,
+	playersUnsorted: PlayerFiltered[],
+): AwardPlayer[] => {
+	return getTopPlayers(options, playersUnsorted).map(getPlayerInfoOffense);
+};
+
+const getTopPlayersDefense = (
+	options: GetTopPlayersOptions,
+	playersUnsorted: PlayerFiltered[],
+): AwardPlayerDefense[] => {
+	return getTopPlayers(options, playersUnsorted).map(getPlayerInfoDefense);
+};
+
+const makeTeams = <T>(
+	players: T[],
+): [
+	{
+		title: "First Team";
+		players: T[];
+	},
+	{
+		title: "Second Team";
+		players: T[];
+	},
+	{
+		title: "Third Team";
+		players: T[];
+	},
+] => {
+	return [
+		{
+			title: "First Team",
+			players: players.slice(0, 5),
+		},
+		{
+			title: "Second Team",
+			players: players.slice(5, 10),
+		},
+		{
+			title: "Third Team",
+			players: players.slice(10, 15),
+		},
+	];
+};
+
+const getRealFinalsMvp = async (
+	players: PlayerFiltered[],
+	champTid: number,
+): Promise<AwardPlayer | undefined> => {
+	const games = await idb.cache.games.getAll(); // Last game of the season will have the two finals teams
+
+	const finalsTids = games[games.length - 1].teams.map(t => t.tid); // Get all playoff games between those two teams - that will be all finals games
+
+	const finalsGames = games.filter(
+		game =>
+			game.playoffs &&
+			finalsTids.includes(game.teams[0].tid) &&
+			finalsTids.includes(game.teams[1].tid),
+	);
+
+	if (finalsGames.length === 0) {
+		return;
+	}
+
+	// Calculate sum of game scores for each player
+	const playerInfos: Map<
+		number,
+		{
+			pid: number;
+			score: number;
+			tid: number;
+			pts: number;
+		}
+	> = new Map();
+
+	for (const game of finalsGames) {
+		for (const t of game.teams) {
+			for (const p of t.players) {
+				const info = playerInfos.get(p.pid) || {
+					pid: p.pid,
+					score: 0,
+					tid: t.tid,
+					pts: 0,
+				};
+
+				// 75% bonus for the winning team
+				const factor = t.tid === champTid ? 1.75 : 1;
+				info.score += factor * p.pts;
+				info.pts += p.pts;
+				playerInfos.set(p.pid, info);
+			}
+		}
+	}
+
+	const playerArray = orderBy(
+		Array.from(playerInfos.values()),
+		"score",
+		"desc",
+	);
+
+	if (playerArray.length === 0) {
+		return;
+	}
+
+	const { pid } = playerArray[0];
+	const p = players.find(p2 => p2.pid === pid);
+
+	if (p) {
+		return {
+			pid: p.pid,
+			name: p.name,
+			tid: p.tid,
+			abbrev: p.abbrev,
+			pts: playerArray[0].pts / finalsGames.length,
+			keyStats: "",
+		};
+	}
+};
+
+export const mvpScore = (p: PlayerFiltered) => {
```

Comment:
Probably would want to use OPS/DPS/GPS in these scores, at least to some extent. But I'm not sure if they're implemented correctly yet, and I haven't seen if they produce "weird" results. But in theory it should work pretty well for offensive players at least. Maybe all players.

Reply:
![image](https://user-images.githubusercontent.com/67447432/107124503-2b22a180-6859-11eb-886d-824c72367b65.png)

OPS seems to be the only one working ATM

Reply:
Yeah I just fixed it locally, making a couple tweaks to the PR branch before I merge it, thanks!